home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/VoteServiceImpl.java

```diff
@@ -0,0 +1,190 @@
+import com.softserveinc.ita.homeproject.homeservice.dto.ReadQuestionVoteDto;
+import com.softserveinc.ita.homeproject.homeservice.dto.ReadVoteDto;
+import com.softserveinc.ita.homeproject.homeservice.exception.BadRequestHomeException;
+import com.softserveinc.ita.homeproject.homeservice.exception.NotFoundHomeException;
+import com.softserveinc.ita.homeproject.homeservice.mapper.ServiceMapper;
+import com.softserveinc.ita.homeproject.homeservice.service.VoteService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class VoteServiceImpl implements VoteService {
+
+    private static final String NOT_FOUND_MESSAGE = "%s with 'id: %d' is not found";
+
+    private static final String POLL_STATUS_VALIDATION_MESSAGE = "Can't create vote on poll with status: '%s'";
+
+    private static final String TRYING_TO_REVOTE_MESSAGE =
+        "You are trying to re-vote on a poll with 'id: %d' that you have already voted";
+
+    private static final String WRONG_QUESTIONS_COUNT_FOR_POLL_MESSAGE =
+        "The number of voted questions does not equal the number of questions in the poll with 'id: %d'";
+
+    private static final String WRONG_QUESTIONS_FOR_POLL_MESSAGE =
+        "Some of the questions you are trying to vote do not match the poll with 'id: %d'";
+
+    private static final String WRONG_ANSWER_COUNT_VALIDATION_MESSAGE =
+        "Wrong count of selected answers to poll question with 'id: %d' (there is should be min 1, max %d)";
+
+    private static final String ANSWER_DOES_NOT_MATCH_QUESTION_VALIDATION_MESSAGE =
+        "The answer variant with 'id: %d' cannot be chosen when voting on the question with 'id: %d'";
+
+    private final VoteRepository voteRepository;
+
+    private final QuestionVoteRepository questionVoteRepository;
+
+    private final PollQuestionRepository questionRepository;
+
+    private final PollRepository pollRepository;
+
+    private final ServiceMapper mapper;
+
+    @Transactional
+    @Override
+    public ReadVoteDto createVote(Long pollId, User currentUser, CreateVoteDto voteDto) {
+        validatePollStatus(pollId);
+        validateReVoting(pollId, currentUser);
+        validateQuestionVotesCount(pollId, voteDto);
+        validatePollQuestionsMatching(pollId, voteDto);
+        validateAnswerCounts(voteDto);
+        validateAnswersMatching(voteDto);
+        var newVote = new Vote();
+        newVote.setPollId(pollId);
+        newVote.setUser(currentUser);
+        var savedVote = voteRepository.save(newVote);
+        List<QuestionVote> savedQuestionVotes = new ArrayList<>();
+        for (CreateQuestionVoteDto dto : voteDto.getQuestionVoteDtos()) {
+            QuestionVote newQuestionVote = mapper.convert(dto, QuestionVote.class);
+            newQuestionVote.setVote(savedVote);
+            var savedQuestionVote = questionVoteRepository.save(newQuestionVote);
```

Comment:
why do we assign the save method to the variable? We dont use it...
Just write savedQuestionVotes.add(questionVoteRepository.save(newQuestionVote)); and it work.

Reply:
done